### PR TITLE
wal should unlock hanged locks

### PR DESCRIPTION
**Pull request by kptfh**

### CODE REVIEW

File: aerospike-storage-backend/src/main/java/com/playtika/janusgraph/aerospike/AerospikeStoreManager.java [hunk 1]
```diff
@@ -80,10 +102,12 @@ public AerospikeStoreManager(Configuration configuration) {
         writeAheadLogCompleter.start();
 
         scanExecutor = new ThreadPoolExecutor(0, configuration.get(SCAN_PARALLELISM),
```
Comment (Aloren) [inline aerospike-storage-backend/src/main/java/com/playtika/janusgraph/aerospike/AerospikeStoreManager.java]: maybe it is better to name thread pools differently? to be able to differentiate scanExecutor from aerospikeExecutor

File: aerospike-storage-backend/src/main/java/com/playtika/janusgraph/aerospike/AerospikeStoreManager.java [hunk 2]
```diff
@@ -176,12 +200,21 @@ public void processAndDeleteTransaction(Value transactionId,
...
+        lockOperations.releaseLocks(locksByStore);
+        deleteWalTransaction(transactionId);
+    }
+
+    private void releaseLocksAndDeleteWalTransaction(Set<Key> keysLocked, Value transactionId) throws BackendException {
+        releaseLocks(keysLocked);
+        deleteWalTransaction(transactionId);
+    }
+
+    void releaseLocks(Set<Key> keysLocked) throws BackendException {
```
Comment (Aloren) [inline aerospike-storage-backend/src/main/java/com/playtika/janusgraph/aerospike/AerospikeStoreManager.java]: you can inline the method

File: aerospike-storage-backend/src/main/java/com/playtika/janusgraph/aerospike/wal/WriteAheadLogCompleter.java [hunk 3]
```diff
@@ -83,13 +93,13 @@ private void completeHangedTransactions(){
...
                     }
-
                     //this is expected behaviour that may have place in case of transaction was interrupted:
                     // - on 'release locks' stage then transaction will fail and just need to release hanged locks
                     // - on 'delete wal transaction' stage and just need to remove transaction
-                    catch (BackendException be) {
-                        logger.info("Failed to complete transaction id={}",
-                                transaction.transactionId, be);
+                    catch (PermanentLockingException be) {
+                        logger.info("Failed to complete transaction id={} as it's already completed", transaction.transactionId, be);
```
Comment (Aloren) [inline aerospike-storage-backend/src/main/java/com/playtika/janusgraph/aerospike/wal/WriteAheadLogCompleter.java]: if this is ok case why do we need to log that on info level? debug should be enough then

File: aerospike-storage-backend/src/test/java/com/playtika/janusgraph/aerospike/LocksGetReleasedAfterFailureTest.java [hunk 4]
```diff
@@ -0,0 +1,229 @@
...
+
+        @Override
+        void deleteWalTransaction(Value transactionId) {
+            if(!fails.get()){
+                super.deleteWalTransaction(transactionId);
+            }
+        }
+    }
+
+    private static class FixedClock extends Clock{
```
Comment (Aloren) [inline aerospike-storage-backend/src/test/java/com/playtika/janusgraph/aerospike/LocksGetReleasedAfterFailureTest.java]: Try `Clock.fixed(instant, zone)`

File: aerospike-storage-backend/src/main/java/com/playtika/janusgraph/aerospike/AerospikeStoreManager.java (hunk 2 above)
Comment (kptfh) [reply aerospike-storage-backend/src/main/java/com/playtika/janusgraph/aerospike/AerospikeStoreManager.java]: it's overridden in test implementation: [FlakingAerospikeStoreManager](https://github.com/Playtika/aerospike-janusgraph-storage-backend/blob/fb0a86a83e6191d7f13b9bc779416edb70ed83ee/aerospike-storage-backend/src/test/java/com/playtika/janusgraph/aerospike/LocksGetReleasedAfterFailureTest.java#L185)

File: aerospike-storage-backend/src/main/java/com/playtika/janusgraph/aerospike/wal/WriteAheadLogCompleter.java (hunk 3 above)
Comment (kptfh) [reply aerospike-storage-backend/src/main/java/com/playtika/janusgraph/aerospike/wal/WriteAheadLogCompleter.java]: yes. It's expected but quite rare corner case. So want to log it with info level to know the frequency of this cases.

**Pull request merged**